File: 717. 1-bit and 2-bit Characters.java
class Solution {
    public boolean isOneBitCharacter(int[] bits) {
        int ind = 0;
        while(ind < bits.length){
            if(bits[ind] == 0){
                ind++;
                if(ind == bits.length) return true;
            }
            else{
                ind += 2;
                if(ind >= bits.length) return false;
            }
        }
        return false;
    }
}

File: 717. 1-bit and 2-bit Characters.c
bool isOneBitCharacter(int* bits, int len) {
    int i = 0;
    while(i < len - 1){
        if(bits[i] == 1){
            i += 2;
        }
        else{
            i++;
        }
    }
    return i == len - 1;
}